### ISSUE TITLE
Streamline arxiv tarball generation

### ISSUE THREAD

**Issue by rodluger** [rodluger/showyourwork#35]
Currently, the entire `src` directory is copied to a temporary folder, which is then pruned of extraneous files and zipped. This is extremely inefficient if there are large datasets in the directory, which get copied over only to be immediately deleted. This can add several seconds -- or even minutes -- to the build step on CI. We should only copy the files we need over to the temporary folder.

**Comment by rodluger** [comment]
@arm61 I think this is the source of the out-of-memory errors you're getting. I'm going to try to fix this today.

**Comment by arm61** [comment]
![](https://media.giphy.com/media/3oeSAz6FqXCKuNFX6o/giphy.gif)

**Comment by rodluger** [comment]
@arm61 Can you pull from `main` and try it out now?

**Comment by arm61** [comment]
No luck, even with building the tarball turned off. It looks (from what I get from the Github action feedback) like it is dying when showyourwork tries to upload the outputs.

**Comment by rodluger** [comment]
It sounds like the issue is with the caching then. Pull from `main` again when you have a chance and, in your `.github/workflows/showyourwork.yml` file, set the `article-cache-number` to `null`:

```yaml
      - name: Build the article PDF
        id: build
        uses: ./showyourwork/showyourwork-action
        with:
          article-cache-number: null
```

And let me know if that causes the build to pass?

**Comment by arm61** [comment]
Amazing, that fixed things! Now how can I help you to sort this so that my builds don't take 25 minutes 🤣 ?

**Comment by rodluger** [comment]
Ok good. One last check: change the article-cache-number to a non-zero
number and run the workflow. If it succeeds, run it again. Let me know if
that works, and if the second one runs quickly.

Sometimes it helps to reset the cache (by bumping that number) to clear the
clutter, especially during dev when file names may be constantly changing.
This can lead to cache sizes that just keep growing with time.

If that doesn't work, I can implement a user option to manually exclude
certain files from the cache. I'm afraid that's the best I can do on my end
since I can't change the GitHub Actions memory limit!

**Comment by arm61** [comment]
Okay so having a non-zero int as the `article-cache-number` stops it working. Which I guess means it is trying to create a cache and failing cause of the size. 

To be honest, 25 minutes isn't too bad for the running and I am happy with it until Github lets me have more memory.

**Comment by rodluger** [comment]
The original issue in this thread was solved; migrating our new conversation to #77.
